### DISCUSSION TITLE
Saved Search returning all fields instead of saved search fields

### DISCUSSION THREAD

**Discussion by netstepinc** [netsuitephp/netsuite-php#132] [Q&A] [answered]
My application is returning verbose responses instead of only the saved search fields.
The documentation I found indicated NetSuite is supposed to return the report columns by default.

I have setSearchPreferences set where argument 3 (returnSearchColumns) = true.
```
$service = new NetSuiteService($config);
$service->setSearchPreferences(false, 1000, true);
```

However, https://github.com/netsuitephp/netsuite-php/blob/master/src/Classes/SearchPreferences.php shows a different order.
```
 static $paramtypesmap = array(
        "bodyFieldsOnly" => "boolean",
        "returnSearchColumns" => "boolean",
        "pageSize" => "integer",
    );
```
**I need to trim the size of the payload way down for speed. What am I missing?**

**How do I use the SearchPreferences class in a saved search request?**

**Comment by jrebs** [comment, marked as answer]
The order in which properties are declared in the `SearchPreferences` (or in any class at all) is immaterial. If you look at the function you're calling, you can verify the correct order of the parameters.

```php
public function setSearchPreferences($bodyFieldsOnly = true, $pageSize = 50, $returnSearchColumns = true)
```

Admittedly I am not sure I've ever worked with saved searches in NetSuite so I don't have an answer for you other than to reassure you that the parameter order is not relevant to the issue. Also, responses do tend to be pretty 'verbose', so without seeing your request and response, it may be difficult for anyone to offer any relevant assistance.
